optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/entity/descriptor/EntityDescriptor.java

```diff
@@ -567,7 +567,15 @@ public long getMaximumValueCount(Solution_ solution, Object entity) {
     public long getProblemScale(Solution_ solution, Object entity) {
         long problemScale = 1L;
         for (GenuineVariableDescriptor<Solution_> variableDescriptor : effectiveGenuineVariableDescriptorList) {
-            problemScale *= variableDescriptor.getValueCount(solution, entity);
+            long valueCount = variableDescriptor.getValueCount(solution, entity);
+            problemScale *= valueCount;
+            if (variableDescriptor.isListVariable()) {
+                // This formula probably makes no sense other than that it results in the same problem scale for both
+                // chained and list variable models.
```

Comment:
optional: add a // TODO with a url to the jira about changing "problem scale" to "search space size"